libraries/AP_InertialSensor/AP_InertialSensor_SCHA63T.cpp

```diff
@@ -0,0 +1,431 @@
+	0xA1, 0xBC, 0x9B, 0x86, 0xD5, 0xC8, 0xEF, 0xF2, 0x49, 0x54, 0x73, 0x6E, 0x3D, 0x20, 0x07, 0x1A,
+	0x6C, 0x71, 0x56, 0x4B, 0x18, 0x05, 0x22, 0x3F, 0x84, 0x99, 0xBE, 0xA3, 0xF0, 0xED, 0xCA, 0xD7,
+	0x35, 0x28, 0x0F, 0x12, 0x41, 0x5C, 0x7B, 0x66, 0xDD, 0xC0, 0xE7, 0xFA, 0xA9, 0xB4, 0x93, 0x8E,
+	0xF8, 0xE5, 0xC2, 0xDF, 0x8C, 0x91, 0xB6, 0xAB, 0x10, 0x0D, 0x2A, 0x37, 0x64, 0x79, 0x5E, 0x43,
+	0xB2, 0xAF, 0x88, 0x95, 0xC6, 0xDB, 0xFC, 0xE1, 0x5A, 0x47, 0x60, 0x7D, 0x2E, 0x33, 0x14, 0x09,
+	0x7F, 0x62, 0x45, 0x58, 0x0B, 0x16, 0x31, 0x2C, 0x97, 0x8A, 0xAD, 0xB0, 0xE3, 0xFE, 0xD9, 0xC4,
+};
+
+AP_InertialSensor_SCHA63T::AP_InertialSensor_SCHA63T(AP_InertialSensor &imu,
+                                                   AP_HAL::OwnPtr<AP_HAL::Device> _dev_accel,
+                                                   AP_HAL::OwnPtr<AP_HAL::Device> _dev_gyro,
+                                                   enum Rotation _rotation)
+    : AP_InertialSensor_Backend(imu)
+    , dev_accel(std::move(_dev_accel))
+    , dev_gyro(std::move(_dev_gyro))
+    , rotation(_rotation)
+{
+}
+
+AP_InertialSensor_Backend *
+AP_InertialSensor_SCHA63T::probe(AP_InertialSensor &imu,
+                                AP_HAL::OwnPtr<AP_HAL::SPIDevice> dev_accel,
+                                AP_HAL::OwnPtr<AP_HAL::SPIDevice> dev_gyro,
+                                enum Rotation rotation)
+{
+    if (!dev_accel || !dev_gyro) {
+        return nullptr;
+    }
+    auto sensor = new AP_InertialSensor_SCHA63T(imu, std::move(dev_accel), std::move(dev_gyro), rotation);
+
+    if (!sensor) {
+        return nullptr;
+    }
+
+#if defined(HAL_GPIO_PIN_RESET)
+    palSetLine(HAL_GPIO_PIN_RESET);
+#endif
+
+    if (!sensor->init()) {
+        delete sensor;
+        return nullptr;
+    }
+
+    return sensor;
+}
+
+void AP_InertialSensor_SCHA63T::start()
+{
+    if (!_imu.register_accel(accel_instance, ACCEL_BACKEND_SAMPLE_RATE, dev_accel->get_bus_id_devtype(DEVTYPE_INS_SCHA63T)) ||
+        !_imu.register_gyro(gyro_instance, GYRO_BACKEND_SAMPLE_RATE,   dev_gyro->get_bus_id_devtype(DEVTYPE_INS_SCHA63T))) {
+        return;
+    }
+
+    // setup sensor rotations from probe()
+    set_gyro_orientation(gyro_instance, rotation);
+    set_accel_orientation(accel_instance, rotation);
+    
+    // setup callbacks
+    dev_accel->register_periodic_callback(1000000UL / ACCEL_BACKEND_SAMPLE_RATE,
+                                          FUNCTOR_BIND_MEMBER(&AP_InertialSensor_SCHA63T::read_fifo_accel, void));
+    dev_gyro->register_periodic_callback(1000000UL / GYRO_BACKEND_SAMPLE_RATE,
+                                         FUNCTOR_BIND_MEMBER(&AP_InertialSensor_SCHA63T::read_fifo_gyro, void));
+}
+
+/*
+  probe and initialise accelerometer
+ */
+bool AP_InertialSensor_SCHA63T::init()
+{
+    dev_accel->get_semaphore()->take_blocking();
+    dev_gyro->get_semaphore()->take_blocking();
+
+    uint8_t v[4];
+
+    hal.scheduler->delay(25);
+
+	RegisterWrite(0, SEL_BANK, 0);
```

Comment:
please run through Tools/CodeStyle/ardupilot-astyle.sh for indentation

Reply:
I fixed it by running this program.